### PR TITLE
Fix ut in gpu mode

### PR DESCRIPTION
**Pull request by Presburger**
#328

### CODE REVIEW

File: thirdparty/faiss/faiss/IndexIVF.cpp [hunk 1]
```diff
@@ -1337,8 +1337,14 @@ void IndexIVF::range_search_preassigned_without_codes(
                 return;
 
             try {
-                InvertedLists::ScopedCodes
-                    scodes(invlists, key, arranged_codes.data());
+                #ifdef USE_GPU
```
Comment (cydrain) [inline thirdparty/faiss/faiss/IndexIVF.cpp:1340]: "#ifdef" should start from the head of this line

File: thirdparty/faiss/faiss/IndexIVF.cpp [hunk 2]
```diff
@@ -1337,8 +1337,14 @@ void IndexIVF::range_search_preassigned_without_codes(
...
+                 auto rol = dynamic_cast<faiss::ReadOnlyArrayInvertedLists*>(invlists);
+                 auto arranged_data = reinterpret_cast<uint8_t*>(rol->pin_readonly_codes->data);
+                 InvertedLists::ScopedCodes scodes(invlists, key, arranged_data);
+                #else
```
Comment (cydrain) [inline thirdparty/faiss/faiss/IndexIVF.cpp:1344]: coding style

File: thirdparty/faiss/faiss/IndexIVF.cpp [hunk 3]
```diff
@@ -1337,8 +1337,14 @@ void IndexIVF::range_search_preassigned_without_codes(
...
+                 InvertedLists::ScopedCodes scodes(invlists, key, arranged_codes.data());
+                #endif
```
Comment (cydrain) [inline thirdparty/faiss/faiss/IndexIVF.cpp:1346]: coding style

File: unittest/test_ivf_nm.cpp [hunk 4]
```diff
@@ -102,13 +102,14 @@ TEST_P(IVFNMTest, ivfnm_basic) {
...
-    AssertVec(result, base_dataset, id_dataset, nq, dim);
-
+    if (index_mode_ == knowhere::IndexMode::MODE_CPU) {
+        auto result = index_->GetVectorById(id_dataset, conf_);
+        AssertVec(result, base_dataset, id_dataset, nq, dim);
+    }
     std::vector<int64_t> ids_invalid(nq, nb);
     auto id_dataset_invalid = knowhere::GenDatasetWithIds(nq, dim, ids_invalid.data());
-    ASSERT_ANY_THROW(index_->GetVectorById(id_dataset_invalid, conf_));
+    if(index_mode_==knowhere::IndexMode::MODE_CPU)
```
Comment (cydrain) [inline unittest/test_ivf_nm.cpp:111]: coding style

File: unittest/benchmark/ref_log/glove/benchmark_knowhere_float/test_knowhere_rhnsw_pq.log [hunk 5]
```diff
@@ -1,46 +0,0 @@
-Running main() from /home/caiyd/vec/knowhere/cmake_build/thirdparty/gtest/googletest-src/googletest/src/gtest_main.cc
```
Comment (cydrain) [inline unittest/benchmark/ref_log/glove/benchmark_knowhere_float/test_knowhere_rhnsw_pq.log:1]: don't remove this file

File: unittest/test_ivf_nm.cpp [hunk 6]
```diff
@@ -102,13 +102,14 @@ TEST_P(IVFNMTest, ivfnm_basic) {
...
     LoadRawData(index_, base_dataset, conf_);
-
-    auto result = index_->GetVectorById(id_dataset, conf_);
-    AssertVec(result, base_dataset, id_dataset, nq, dim);
-
+    if (index_mode_ == knowhere::IndexMode::MODE_CPU) {
+        auto result = index_->GetVectorById(id_dataset, conf_);
+        AssertVec(result, base_dataset, id_dataset, nq, dim);
+    }
     std::vector<int64_t> ids_invalid(nq, nb);
```
Comment (cydrain) [inline unittest/test_ivf_nm.cpp:109]: you can embrace all these lines into "if (index_mode_ == knowhere::IndexMode::MODE_CPU)“

Comment (Presburger) [reply unittest/test_ivf_nm.cpp:109]: ok

File: unittest/test_ivf_nm.cpp [hunk 7]
```diff
@@ -91,6 +91,9 @@ LoadRawData(const knowhere::VecIndexPtr index,
 }
 
 TEST_P(IVFNMTest, ivfnm_basic) {
+    if (index_mode_ != knowhere::IndexMode::MODE_CPU)
```
Comment (cydrain) [inline unittest/test_ivf_nm.cpp:94]: why disable this?

**Pull request merged**